### PR TITLE
Allow email change

### PR DESCRIPTION
**Pull request by marceloschmidt**

### CODE REVIEW

**Comment by marceloschmidt** [comment]
Closes #1462

File: packages/rocketchat-lib/server/functions/checkEmailAvailability.coffee [hunk 1]
```diff
@@ -0,0 +1,2 @@
+RocketChat.checkEmailAvailability = (email) ->
+	return not Meteor.users.findOne({ "emails.address": { $regex : new RegExp("^" + s.trim(email) + "$", "i") } })
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/functions/checkEmailAvailability.coffee]: Should we scape regex here?

File: packages/rocketchat-lib/server/methods/setEmail.coffee [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		user = Meteor.user()
+
+		if not RocketChat.settings.get("Accounts_AllowEmailChange")
+			throw new Meteor.Error(403, "[methods] setEmail -> E-mail change not allowed")
+
+		if user.emails?[0]?.address is email
+			return email
+
+		emailValidation = /^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/
+		if not emailValidation.test email
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/setEmail.coffee]: Why validate and check availability two times?

File: packages/rocketchat-ui-account/account/accountProfile.html [hunk 3]
```diff
@@ -19,9 +19,19 @@
...
-						    <input type="text" name="username" id="username" placeholder="{{username}}" disabled="disabled" title="{{usernameChangeDisabled}}" />
+							<input type="text" name="username" id="username" placeholder="{{username}}" disabled="disabled" title="{{usernameChangeDisabled}}" />
+						{{/if}}
+						</div>
+					</div>
+					<div class="input-line">
+						<label for="email">{{_ "E-mail"}}</label>
+						<div>
+						{{#if allowEmailChange}}
+							<input type="text" name="email" id="email" placeholder="{{email}}" />
```
Comment (rodrigok) [inline packages/rocketchat-ui-account/account/accountProfile.html]: Should use type **email**

File: packages/rocketchat-ui-admin/admin/users/adminUserEdit.html [hunk 4]
```diff
@@ -13,6 +13,10 @@
 					<label for="username">{{_ "Username"}}</label>
 					<input type="text" id="username" autocomplete="off" value="{{username}}">
 				</div>
+				<div class="input-line">
+					<label for="email">{{_ "E-mail"}}</label>
+					<input type="text" id="email" autocomplete="off" value="{{email}}">
```
Comment (rodrigok) [inline packages/rocketchat-ui-admin/admin/users/adminUserEdit.html]: Should use type **email**

File: packages/rocketchat-lib/server/methods/setEmail.coffee (hunk 2 above)
Comment (marceloschmidt) [inline packages/rocketchat-lib/server/methods/setEmail.coffee]: Are you talking about once on method and another on function? Usually on functions we don't return an error, just false for end of execution. Functions are intended to be run by methods or by api calls, AFAICT. Let's discuss with the group if that's the real intention...

**Comment by geekgonecrazy** [comment]
LGTM

Tested this and works like a charm :+1:

**Comment by rodrigok** [comment]
LGTM

**Pull request merged**